src/System.Windows.Forms/src/System/Windows/Forms/LinkLabel.Link.LinkLabelLinkAccessibleObject.cs

```diff
@@ -0,0 +1,179 @@
+                    {
+                        return _owningLink.Description;
+                    }
+                }
+
+                public override string? Name
+                {
+                    get
+                    {
+                        string? name = _owningLink.Owner.Text;
+
+                        // Return the full name of the link label
+                        // as sometimes the link name in isolation
+                        // is unusable when using a screen reader.
+                        if (_owningLink.Owner.UseMnemonic)
+                        {
+                            name = WindowsFormsUtils.TextWithoutMnemonics(name);
+                        }
+
+                        return name;
+                    }
+                    set => base.Name = value;
+                }
+
+                public override AccessibleObject Parent
+                {
+                    get
+                    {
+                        return _owningLink.Owner.AccessibilityObject;
+                    }
+                }
+
+                public override AccessibleRole Role
+                {
+                    get
+                    {
+                        return AccessibleRole.Link;
+                    }
+                }
+
+                public override AccessibleStates State
+                {
+                    get
+                    {
+                        AccessibleStates state = AccessibleStates.Focusable;
+
+                        // Selected state
+                        //
+                        if (_owningLink.Owner.FocusLink == _owningLink)
+                        {
+                            state |= AccessibleStates.Focused;
+                        }
+
+                        return state;
+                    }
+                }
+
+                public override string Value
+                {
+                    get
+                    {
+                        // Narrator announces Link's text twice, once as a Name property and once as a Value, thus removing value.
+                        // Value is optional for this role (Link).
+                        return string.Empty;
+                    }
+                }
+
+                public override void DoDefaultAction()
+                {
+                    _owningLink.Owner.OnLinkClicked(new LinkLabelLinkClickedEventArgs(_owningLink));
+                }
+
+                internal override bool IsIAccessibleExSupported() => true;
+
+                internal override object? GetPropertyValue(UiaCore.UIA propertyID)
+                {
+                    if (propertyID == UiaCore.UIA.IsEnabledPropertyId)
+                    {
+                        if (!_owningLink.Owner.Enabled)
+                        {
+                            return false;
+                        }
+                    }
+
+                    return base.GetPropertyValue(propertyID);
+                }
```

Comment:
Reworking to the expression will be helpful in the future:
```suggestion
                internal override object? GetPropertyValue(UiaCore.UIA propertyID)
                    => propertyID switch
                    {
                        UiaCore.UIA.IsEnabledPropertyId => _owningLinkLabel.Enabled,
                        _ => base.GetPropertyValue(propertyID)
                    };
```